js/coinbase.js

```diff
@@ -1633,6 +1638,165 @@ module.exports = class coinbase extends Exchange {
+        const stopLossPrice = this.safeNumber (params, 'stopLossPrice');
+        const takeProfitPrice = this.safeNumber (params, 'takeProfitPrice');
+        const isStop = stopPrice !== undefined;
+        const isStopLoss = stopLossPrice !== undefined;
+        const isTakeProfit = takeProfitPrice !== undefined;
+        const timeInForce = this.safeString (params, 'timeInForce');
+        const postOnly = (timeInForce === 'PO') ? true : this.safeValue2 (params, 'postOnly', 'post_only', false);
+        let stopDirection = this.safeString (params, 'stop_direction');
+        if (type === 'limit') {
+            if (isStop) {
+                if (stopDirection === undefined) {
+                    stopDirection = (side === 'buy') ? 'STOP_DIRECTION_STOP_DOWN' : 'STOP_DIRECTION_STOP_UP';
+                }
+                request['order_configuration'] = {
+                    'stop_limit_stop_limit_gtc': {
+                        'base_size': this.amountToPrecision (symbol, amount),
+                        'limit_price': this.priceToPrecision (symbol, price),
+                        'stop_price': this.priceToPrecision (symbol, stopPrice),
+                        'stop_direction': stopDirection,
+                    },
+                };
+            } else if (isStopLoss || isTakeProfit) {
+                let triggerPrice = undefined;
+                if (isStopLoss) {
+                    if (stopDirection === undefined) {
+                        stopDirection = (side === 'buy') ? 'STOP_DIRECTION_STOP_UP' : 'STOP_DIRECTION_STOP_DOWN';
+                    }
+                    triggerPrice = this.priceToPrecision (symbol, stopLossPrice);
+                } else {
+                    if (stopDirection === undefined) {
+                        stopDirection = (side === 'buy') ? 'STOP_DIRECTION_STOP_DOWN' : 'STOP_DIRECTION_STOP_UP';
+                    }
+                    triggerPrice = this.priceToPrecision (symbol, takeProfitPrice);
+                }
+                request['order_configuration'] = {
+                    'stop_limit_stop_limit_gtc': {
+                        'base_size': this.amountToPrecision (symbol, amount),
+                        'limit_price': this.priceToPrecision (symbol, price),
+                        'stop_price': triggerPrice,
+                        'stop_direction': stopDirection,
+                    },
+                };
+            } else {
+                request['order_configuration'] = {
+                    'limit_limit_gtc': {
+                        'base_size': this.amountToPrecision (symbol, amount),
+                        'limit_price': this.priceToPrecision (symbol, price),
+                        'post_only': postOnly,
+                    },
+                };
+            }
+        } else {
+            if (isStop || isStopLoss || isTakeProfit) {
+                throw new NotSupported (this.id + ' createOrder() only stop limit orders are supported');
+            }
```

Comment:
Should we validate the required `market_market_ioc` parameters?

`quote_size` string Amount of quote currency to spend on order. **Required for BUY orders**.

`base_size` string Amount of base currency to spend on order. **Required for SELL orders**.